tests/server/test_desired_state_versions.py

```diff
@@ -20,17 +20,112 @@
+from inmanta.data.model import DesiredStateLabel
+from inmanta.protocol import SessionEndpoint, methods
+from inmanta.server import SLICE_AGENT_MANAGER, config, protocol
+from inmanta.types import Apireturn
+from utils import retry_limited
+
+
+class BaseAgent(SessionEndpoint):
+
+    def __init__(
+        self,
+        environment: Optional[uuid.UUID] = None,
+    ):
+        """
+        :param environment: environment id
+        """
+        super().__init__(name="agent", timeout=server_timeout.get(), reconnect_delay=agent_reconnect_delay.get())
+        self._env_id = environment
+        self.enabled: dict[str, bool] = {}
+        self.limit = 100
+
+    async def start_connected(self) -> None:
+        """
+        Setup our single endpoint
+        """
+        await self.add_end_point_name(AGENT_SCHEDULER_ID)
+
+    @protocol.handle(methods.set_state)
+    async def set_state(self, agent: Optional[str], enabled: bool) -> Apireturn:
+        self.enabled[agent] = enabled
+
+        return 200
+
+    async def on_reconnect(self) -> None:
+        pass
+
+    async def on_disconnect(self) -> None:
+        pass
+
+    @protocol.handle(methods.trigger, env="tid", agent="id")
+    async def trigger_update(self, env: uuid.UUID, agent: str, incremental_deploy: bool) -> Apireturn:
+        return 200
+
+    @protocol.handle(methods.trigger_read_version, env="tid", agent="id")
+    async def read_version(self, env: uuid.UUID) -> Apireturn:
+        await data.Scheduler._execute_query(
+            f"""
+                                   INSERT INTO {data.Scheduler.table_name()}
+                                   VALUES($1, NULL)
+                                   ON CONFLICT DO NOTHING
+                               """,
+            env,
+        )
+
+        items = await ConfigurationModel.get_list(environment=env, released=True, order_by_column="version", order="DESC")
+        sched = await Scheduler.get_one(environment=env)
+        last_version = sched.last_processed_model_version or 0
+        for cm in items:
+            if cm.version > self.limit:
+                continue
+            if cm.version < last_version:
+                return 200
+            await Scheduler.set_last_processed_model_version(env, cm.version)
+            return 200
+        return 200
+
+    @protocol.handle(methods.do_dryrun, env="tid", dry_run_id="id")
+    async def run_dryrun(self, env: uuid.UUID, dry_run_id: uuid.UUID, agent: str, version: int) -> Apireturn:
+        return 200
+
+    @protocol.handle(methods.get_parameter, env="tid")
+    async def get_facts(self, env: uuid.UUID, agent: str, resource: dict[str, Any]) -> Apireturn:
+        return 200
+
+    @protocol.handle(methods.get_status)
+    async def get_status(self) -> Apireturn:
+        return 200, {}
+
+
+@pytest.fixture(scope="function")
+async def base_agent(server, environment):
+    """Construct an agent that does nothing"""
```

Comment:
This is the same docstring as the `null_agent` fixture. Why would I use this fixture instead of the `null_agent` fixture?